tools/cmgen/src/ProgressUpdater.cpp

```diff
@@ -26,18 +26,40 @@ static void moveCursorUp(size_t n) {
 }
 
 static void showCursor() {
-    signal(SIGINT, SIG_DFL);
-    std::cout << "\033[?25h" << std::flush;
+    struct sigaction sa;
+    sa.sa_handler = SIG_DFL;
+    sigemptyset(&sa.sa_mask);
+    sa.sa_flags = 0;
+    sigaction(SIGINT, &sa, nullptr);
```

Comment:
it seems like you could factorize all this code into a `signal_safe()` function. instead of repeating 3 times.

Reply:
Fixed